### DISCUSSION TITLE
Debugging and hot reloading for developing extensions

### DISCUSSION THREAD

**Discussion by 60-gh12** [vladmandic/sdnext#1097] [Q&A] [answered]
Hi,

I'm trying to work on my own extensions/updates to existing extensions.

I am new to python development and I have a couple questions. Firstly, how can I run the portal in a debug mode that would allow me to step through the code and stop at breakpoints etc? Ideally I would like to do this in VS code.

Second, as I am making updates to an extension, is it possible to somehow hot reload the changes made to that extension so that I do not have to completely reload the portal every time I make a change? Perhaps by using something like this: https://github.com/breuleux/jurigged

Thanks!

**Comment by vladmandic** [comment]
hot reloading is experimental at best and not trivial.
but check out <https://reloadium.io/>

**Reply by 60-gh12** [reply]
Ok, thank you. I got the debugging working at least. I didn't realize I had to set the extension as a submodule and then set the breakpoints there. I had been working on the extension in a separate vscode workspace.

I will check out reloadium!

**Reply by 60-gh12** [reply, marked as answer]
If anyone is wondering. I got the hot reloading working with jurigged (https://github.com/breuleux/jurigged)

I installed jurigged in my venv and then modified the "launch" line in the webui.bat to look like this:

```
:launch
%PYTHON% -m jurigged --watch <file-path-to-extension> -v launch.py %*
```

Now obviously this does not reload UI elements, only changes the functions that get called, but this was enough for me to get started!
